pkg/egresspolicy/egresspolicy.go

```diff
@@ -0,0 +1,566 @@
+}
+
+// policyConfigSelectsPod determines if the given pod is selected by the policy
+// config based on matching labels of config and pod.
+func (config *EgressPolicyConfig) policyConfigSelectsPod(podInfo *podMetadata) bool {
+	labelsToMatch := labels.Set(podInfo.labels)
+	nsLabels := labels.Set(podInfo.nsLabels)
+	for k, v := range nsLabels {
+		labelsToMatch[policy.JoinPath(k8sConst.PodNamespaceMetaLabels, k)] = v
+	}
+	for _, selector := range config.endpointSelectors {
+		if selector.Matches(labelsToMatch) {
+			return true
+		}
+	}
+	return false
+}
+
+func Parse(cenp *v2alpha1.CiliumEgressNATPolicy) (*EgressPolicyConfig, error) {
+	var endpointSelectorList []api.EndpointSelector
+	var dstCidrList []*net.IPNet
+
+	allowAllNamespacesRequirement := slim_metav1.LabelSelectorRequirement{
+		Key:      k8sConst.PodNamespaceLabel,
+		Operator: slim_metav1.LabelSelectorOpExists,
+	}
+	name := cenp.ObjectMeta.Name
+
+	if name == "" {
+		return nil, fmt.Errorf("CiliumEgressNATPolicy must have a name")
+	}
+
+	for _, cidrString := range cenp.Spec.DestinationCIDRs {
+		_, cidr, err := net.ParseCIDR(cidrString)
+		if err != nil {
+			log.WithError(err).Warnf("CiliumEgressNATPolicy %v error parsing cidr", name)
+			continue
+		}
+		dstCidrList = append(dstCidrList, cidr)
+	}
+
+	for _, egressRule := range cenp.Spec.Egress {
+		if egressRule.NamespaceSelector != nil {
+			prefixedNsSelector := egressRule.NamespaceSelector
+			matchLabels := map[string]string{}
+			// We use our own special label prefix for namespace metadata,
+			// thus we need to prefix that prefix to all NamespaceSelector.MatchLabels
+			for k, v := range egressRule.NamespaceSelector.MatchLabels {
+				matchLabels[policy.JoinPath(k8sConst.PodNamespaceMetaLabels, k)] = v
+			}
+			egressRule.NamespaceSelector.MatchLabels = matchLabels
+
+			// We use our own special label prefix for namespace metadata,
+			// thus we need to prefix that prefix to all NamespaceSelector.MatchLabels
+			for i, lsr := range egressRule.NamespaceSelector.MatchExpressions {
+				lsr.Key = policy.JoinPath(k8sConst.PodNamespaceMetaLabels, lsr.Key)
+				egressRule.NamespaceSelector.MatchExpressions[i] = lsr
+			}
+
+			// Empty namespace selector selects all namespaces (i.e., a namespace
+			// label exists).
+			if len(egressRule.NamespaceSelector.MatchLabels) == 0 && len(egressRule.NamespaceSelector.MatchExpressions) == 0 {
+				egressRule.NamespaceSelector.MatchExpressions = []slim_metav1.LabelSelectorRequirement{allowAllNamespacesRequirement}
+			}
+
+			endpointSelectorList = append(
+				endpointSelectorList,
+				api.NewESFromK8sLabelSelector("", prefixedNsSelector, egressRule.PodSelector))
+		} else if egressRule.PodSelector != nil {
+			endpointSelectorList = append(
+				endpointSelectorList,
+				api.NewESFromK8sLabelSelector("", egressRule.PodSelector))
+		} else {
+			return nil, fmt.Errorf(
+				"CiliumEgressNATPolicy %v cannot have both nil namespace selector and nil pod selector",
+				name)
```

Comment:
No need to format the error with the name. The caller will log the name of the resource that failed.

Reply:
good point. fixed

Reply:
good point. fixed